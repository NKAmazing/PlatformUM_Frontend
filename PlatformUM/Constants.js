export const categoriesData = [
    {
        title: 'Location',
        image: require('./assets/images/location.png')
    },
    {
        title: "Today's Trip!",
        image: require('./assets/images/bus.png')
    },
    {
        title: 'Companies',
        image: require('./assets/images/company.png')
    }
]