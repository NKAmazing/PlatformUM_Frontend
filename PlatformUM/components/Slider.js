import React from 'react';
import { View, Image, StyleSheet, Dimensions } from 'react-native';
import Swiper from 'react-native-swiper';

const { width } = Dimensions.get('window');


const Slider = () => {
  return (
    <View style={styles.wrapper}>
        <Swiper style={styles.swiper} showsButtons={false} showsPagination={true}>
        <View style={styles.slide}>
            <Image
            source={require('../assets/images/bus1.jpg')}
            style={styles.image}
            />
        </View>
        <View style={styles.slide}>
            <Image
            source={require('../assets/images/bus2.jpg')}
            style={styles.image}
            />
        </View>
        <View style={styles.slide}>
            <Image
            source={require('../assets/images/bus3.jpg')}
            style={styles.image}
            />
        </View>
        </Swiper>
    </View>
  );
};

const styles = StyleSheet.create({
    wrapper: {
        marginTop: '8%',
        marginBottom: '8%',
        height: 200,
    },
    swiper: {},
    slide: {
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
    },
    image: {
      width: width-50,
      flex: 1,
      borderRadius: 15,
    },
  });

export default Slider;