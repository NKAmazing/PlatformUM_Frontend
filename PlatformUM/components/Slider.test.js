import React from 'react';
import renderer from 'react-test-renderer';
import Slider from './Slider';
import { testMessages } from '../Constants';


test(testMessages.slider, () => {
  const tree = renderer.create(<Slider />).toJSON();
  expect(tree).toMatchSnapshot();
});
