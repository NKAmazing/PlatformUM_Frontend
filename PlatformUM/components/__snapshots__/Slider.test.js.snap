// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider component renders correctly 1`] = `
<View
  style={
    {
      "height": 200,
      "marginBottom": "8%",
      "marginTop": "8%",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "backgroundColor": "transparent",
          "flex": 1,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      autoplay={false}
      autoplayDirection={true}
      autoplayTimeout={2.5}
      bounces={false}
      contentContainerStyle={
        [
          {
            "backgroundColor": "transparent",
          },
          {},
        ]
      }
      contentOffset={
        {
          "x": 0,
        }
      }
      disableNextButton={false}
      disablePrevButton={false}
      horizontal={true}
      index={0}
      loadMinimal={false}
      loadMinimalSize={1}
      loop={true}
      onIndexChanged={[Function]}
      onMomentumScrollEnd={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollsToTop={false}
      showsButtons={false}
      showsHorizontalScrollIndicator={false}
      showsPagination={true}
      showsVerticalScrollIndicator={false}
    >
      <View>
        <View
          style={
            [
              {
                "height": 1334,
                "width": 750,
              },
              {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={1}
              style={
                {
                  "borderRadius": 15,
                  "flex": 1,
                  "width": 700,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            [
              {
                "height": 1334,
                "width": 750,
              },
              {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={1}
              style={
                {
                  "borderRadius": 15,
                  "flex": 1,
                  "width": 700,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            [
              {
                "height": 1334,
                "width": 750,
              },
              {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={1}
              style={
                {
                  "borderRadius": 15,
                  "flex": 1,
                  "width": 700,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            [
              {
                "height": 1334,
                "width": 750,
              },
              {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={1}
              style={
                {
                  "borderRadius": 15,
                  "flex": 1,
                  "width": 700,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            [
              {
                "height": 1334,
                "width": 750,
              },
              {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={1}
              style={
                {
                  "borderRadius": 15,
                  "flex": 1,
                  "width": 700,
                }
              }
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      pointerEvents="none"
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "bottom": 25,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#007aff",
              "borderRadius": 4,
              "height": 8,
              "marginBottom": 3,
              "marginLeft": 3,
              "marginRight": 3,
              "marginTop": 3,
              "width": 8,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "rgba(0,0,0,.2)",
              "borderRadius": 4,
              "height": 8,
              "marginBottom": 3,
              "marginLeft": 3,
              "marginRight": 3,
              "marginTop": 3,
              "width": 8,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "rgba(0,0,0,.2)",
              "borderRadius": 4,
              "height": 8,
              "marginBottom": 3,
              "marginLeft": 3,
              "marginRight": 3,
              "marginTop": 3,
              "width": 8,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
